okta/okta.go

```diff
@@ -101,6 +101,8 @@ type userManager struct {
 	reconciliateID func(id string) (newID string)
 }
 
+var ErrNeedsReauthentication = errors.New("user session has expired so they must be reauthenticated")
```

Comment:
I'm using this in `middleware_auth_user.go` if you're using API key headers, because it will attempt to check the API key and respond with an error if you don't return some error. Mostly the `ErrNeedsReauthorization` is for consistency between `GetUserByToken` and `GetUserByID`. If you want, I could just add the `needsReauth` bool to `GetUserByID` but don't have strong feelings about it.

Reply:
In that case wouldn't it be better to just return the error, and have the caller check for the error type?

Reply:
I think I originally needed a bool but now don't need it, so I just changed to an errors.Cause check.